src/ast/transform/SemanticChecker.cpp

```diff
@@ -610,6 +611,25 @@ void SemanticCheckerImpl::checkRelation(const Relation& relation) {
         }
     }
 
+    // check subsumption relations
+    bool hasSubsumptiveRule = visitExists(program, [&](const ast::SubsumptiveClause& sClause) {
+        return sClause.getHead()->getQualifiedName() == relation.getQualifiedName();
+    });
+    if (relation.getRepresentation() == RelationRepresentation::BTREE_DELETE && !hasSubsumptiveRule) {
+        report.addWarning("No subsumptive rule for relation " + toString(relation.getQualifiedName()),
+                relation.getSrcLoc());
+    } else if (relation.getRepresentation() != RelationRepresentation::BTREE_DELETE && hasSubsumptiveRule) {
```

Comment:
Q: How difficult is it to auto-add `btree_delete` property to the relation if using subsumption for that relation, and is there any downside in expressibility (e.g. use different data structure) in doing so?

Reply:
We leave this to a later version - I agree it would be great to have a small transformer doing this for the users.